server/apps/publish/archive_publish.py

```diff
@@ -282,48 +302,92 @@ def publish(self, doc, updates, target_output_channels=None):
                 updates['source'] = desk['source'] if desk and desk.get('source', '') \
                     else DEFAULT_SOURCE_VALUE_FOR_MANUAL_ARTICLES
 
+            updates['pubstatus'] = 'canceled' if self.publish_type == 'killed' else 'usable'
+            updated.update(updates)
+
+        # Step 2(a)
+        if self.published_state == 'killed':
+            queued_items = get_resource_service('publish_queue').get(
+                req=None, lookup={'item_id': updated[config.ID_FIELD]})
+
+            if queued_items.count():
+                # Step 2(a)(i)
+                subscribers = self._find_subscribers(queued_items)
+                recipients = [s['email'] for s in subscribers]
+                send_article_killed_email(doc, recipients)
+
+                # Step 2(a)(ii)
+                no_formatters, queued = self.queue_transmission(updated, subscribers, None)
+        elif self.published_state == 'corrected':  # Step 2(b)
+            query = {'$and': [{'item_id': updated[config.ID_FIELD]}, {'publishing_action': 'published'}]}
+            queued_items = get_resource_service('publish_queue').get(req=None, lookup=query)
+
+            if queued_items.count():
+                # Step 2(b)(i)
+                subscribers = self._find_subscribers(queued_items)
+                no_formatters, queued = self.queue_transmission(updated, subscribers)
+
+                # Step 2(b)(ii)
+                active_subscribers = list(get_resource_service('subscribers').get(req=None, lookup={'is_active': True}))
+                subscribers_yet_to_receive = []
+
+                for subscriber in active_subscribers:
+                    found = False
+                    for s in subscribers:
+                        if s[config.ID_FIELD] == subscriber[config.ID_FIELD]:
+                            found = True
+                            break
+
+                    if not found:
+                        subscribers_yet_to_receive.append(subscriber)
+
+                if len(subscribers_yet_to_receive):
```

Comment:
Perhaps make it more explicit by comparing the len to `> 0`...
(it would also make it consistent with a similar comparison a few lines below)